code/parachain/frame/composable-traits/src/assets.rs

```diff
@@ -19,6 +20,76 @@ pub const ASSET_METADATA_SYMBOL_LENGTH: usize = 16;
 pub type BiBoundedAssetName = BiBoundedVec<u8, 1, ASSET_METADATA_NAME_LENGTH>;
 pub type BiBoundedAssetSymbol = BiBoundedVec<u8, 1, ASSET_METADATA_SYMBOL_LENGTH>;
 
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+#[serde(rename_all = "snake_case")]
+#[cfg_attr(feature = "std", derive(JsonSchema, QueryResponses))]
+pub enum ExecuteMsg {
+	// denom is asset u128 to string, for example PICA on Picasso is "1"
+	// all admins default to contract address if not specified
+	#[cfg_attr(feature = "std", returns(CreateResponse))]
+	Create {
```

Comment:
Why everything is optional? Why not create multiple create instead? Too generic opening the chance for messages that don't make sense. 
Validation becomes more difficult

Reply:
We support no_std, so something optional.

No sure what you mean about?

Reply:
If somebodu wants  Rust code they generate rust from Json Schema with what ever dependencies they want

Reply:
I support XCVM + Precompiles with in no_std env.

Reply:
btw i moved composable std  files in cosmwasm core lib, so all CW will be here for sharing https://github.com/ComposableFi/composable/pull/3774